.github/workflows/prod-deploy-worker.yml

```diff
@@ -0,0 +1,117 @@
+name: Deploy PROD Worker
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  workflow_dispatch:
+
+jobs:
+  test_worker:
+    strategy:
+      matrix:
+        name: ['novu/worker', 'novu/worker-ee']
+    uses: ./.github/workflows/reusable-worker-e2e.yml
+    with:
+      ee: ${{ contains (matrix.name,'ee') }}
+
+  build_prod_image:
+    if: "!contains(github.event.head_commit.message, 'ci skip')"
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    needs: test_worker
+    timeout-minutes: 80
+    environment: Production
+    outputs:
+      docker_image: ${{ steps.build-image.outputs.IMAGE }}
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write
+    strategy:
+      matrix:
+        name: ['novu/worker', 'novu/worker-ee']
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-project
+
+      - name: build api
+        run: pnpm build:worker
+
+      - name: Setup Depot
+        uses: depot/setup-action@v1
+        with:
+          oidc: true
+
+      - name: Set Bull MQ Env variable for EE
+        shell: bash
+        run: |
+          echo "BULL_MQ_PRO_NPM_TOKEN=${{ secrets.BULL_MQ_PRO_NPM_TOKEN }}" >> $GITHUB_ENV
+        if: ${{contains(matrix.name, 'ee')}}
+
+      - name: Build, tag, and push image to Amazon ECR
+        id: build-image
+        env:
+          REGISTRY_OWNER: novuhq
+          DOCKER_NAME: ${{ matrix.name }}
+          IMAGE_TAG: ${{ github.sha }}
+          GH_ACTOR: ${{ github.actor }}
+          GH_PASSWORD: ${{ secrets.GH_PACKAGES }}
+          DEPOT_PROJECT_ID: 6sj0jfv0n7
+        run: |
+          echo $GH_PASSWORD | docker login ghcr.io -u $GH_ACTOR --password-stdin
+          cd apps/worker && pnpm --silent --workspace-root pnpm-context -- apps/worker/Dockerfile | depot build --build-arg BULL_MQ_PRO_TOKEN=${BULL_MQ_PRO_NPM_TOKEN} --build-arg PACKAGE_PATH=apps/worker - -t novu-worker --load
+          docker tag novu-worker ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:latest
+          docker tag novu-worker ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:prod
+          docker tag novu-worker ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG
+
+          docker run --network=host --name worker -dit --env NODE_ENV=test ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG
+          docker run --network=host appropriate/curl --retry 10 --retry-delay 5 --retry-connrefused http://localhost:3004/v1/health-check | grep 'ok'
+
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:prod
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:latest
+          docker push ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG
+          echo "::set-output name=IMAGE::ghcr.io/$REGISTRY_OWNER/$DOCKER_NAME:$IMAGE_TAG"
+
+  deploy_prod_worker_eu:
+    if: contains( ${{ matrix.name }} , 'ee')
```

Comment:
This is not a valid syntax from github actions and will fail when run. We have a similar problem in prod pipelines, there I hot fixed it by disabling the matrix above until we find a long term solution

Reply:
fixed